src/mysql.erl

```diff
@@ -729,10 +733,18 @@ handle_call(start_transaction, _From,
                                                ?cmd_timeout),
     SockMod:setopts(Socket, [{active, once}]),
     State1 = update_state(Res, State),
-    {reply, ok, State1#state{transaction_level = L + 1}};
-handle_call(rollback, _From, State = #state{socket = Socket, sockmod = SockMod,
-                                            status = Status, transaction_level = L})
+    {reply, ok, State1#state{transaction_level = L + 1, monitors = maps:put(FromPid, MRef, Monitors)}};
+handle_call(rollback, {FromPid, _}, State = #state{socket = Socket, sockmod = SockMod,
+                                            status = Status, transaction_level = L, monitors = Monitors})
   when Status band ?SERVER_STATUS_IN_TRANS /= 0, L >= 1 ->
+    NewMonitors = case maps:take(FromPid, Monitors) of
```

Comment:
If one process does a nested transaction, we shouldn't demonitor when the inner transaction finishes. Perhaps we can use a simple list of monitors where we add one regardless of whether there already is one for the process. Then we pop them and demonitor in commit/rollback. An other solution would be to add a counter per process and demonitor only when the counter reaches zero. (I.e. `[{Pid, {Count, MonitorRef}}]` if we use an orddict.)

It's the same code for rollback and commit, so it should be moved to a helper function.

Note the recommended line length of 80 as well.

Reply:
Yes I'll keep under loop all your observations. List works ok.

I'll do the changes and when you give me the ok I'll write the test case as well.

Reply:
How do you prefer:

Solution 1:

- In the monitor list we add: `[{FromPid, MRef} | Monitors]}`
- in rollback and commit we do:

`handle_call(rollback, {FromPid, _}, State = #state{..., monitors = [{FromPid, MRef}|NewMonitors]})`

Solution 2:

- In the monitor list we add: `[Mref | Monitors]}`
- in rollback and commit we do:
`handle_call(rollback, _From, State = #state{..., monitors = [MRef|NewMonitors]})`

Idea is that having PID in the list it's only as a additional safe check that our logic is ok .. nothing more..

Reply:
I like the Pid. It causes a crash if the wrong pid is calling commit/rollback, which is good.

Perhaps we can get rid of transaction_level. We can check if Monitors is empty instead.

Reply:
Please check and let me know if you want to to the transaction_level change as well.

Reply:
I see that the level is used to generate breakpoint names. We can keep it.